src/dispatch/participant/service.py

```diff
@@ -79,34 +79,43 @@ def get_all_by_incident_id(*, db_session, incident_id: int) -> List[Optional[Par
 def get_or_create(
     *,
     db_session,
-    incident_id: int,
+    subject_id: int,
+    subject_type: str,
     individual_id: int,
     service_id: int,
     participant_roles: List[ParticipantRoleCreate],
 ) -> Participant:
     """Gets an existing participant object or creates a new one."""
-    from dispatch.incident import service as incident_service
-
-    participant = (
+    participant: Participant = (
         db_session.query(Participant)
-        .filter(Participant.incident_id == incident_id)
+        .filter(
+            Participant.incident_id == subject_id
+            if subject_type == "incident"
+            else Participant.case_id == subject_id
+        )
         .filter(Participant.individual_contact_id == individual_id)
         .one_or_none()
     )
 
     if not participant:
-        incident = incident_service.get(db_session=db_session, incident_id=incident_id)
+        if subject_type == "incident":
+            from dispatch.incident import service as incident_service
 
-        # We get information about the individual
-        individual_contact = individual_service.get(
-            db_session=db_session, individual_contact_id=individual_id
-        )
+            subject = incident_service.get(db_session=db_session, incident_id=subject_id)
+        if subject_type == "case":
+            from dispatch.case import service as case_service
```

Comment:
Is this causing you circular import issues?

Reply:
No, I moved the imports (and there's still no circular import) to the top of the file since that pattern isn't necessary.